src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt

```diff
@@ -0,0 +1,27 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import jakarta.transaction.Transactional
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.hdc.HdcApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.hdc.HdcLicenceData
+
+@Service
+class HdcService(
+  private val hdcApiClient: HdcApiClient,
+) {
+
+  @Transactional
+  fun getHdcLicenceData(bookingId: String?): HdcLicenceData? {
+    if (bookingId == null) {
+      return null
+    }
+
+    val licenceData = this.hdcApiClient.getByBookingId(bookingId.toLong())
```

Comment:
**suggestion:** This should probably be bound to a long in the controller parameter.
This would then be converted to a 400 bad request if its not parseable

(Generally worth pushing any conversion/marshalling to the boundaries of the application - fail fast rather than potentially having to deal with bad data in midst of business logic)

Reply:
Worth thinking of the error handling [here](https://github.com/ministryofjustice/hmpps-hdc-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt#L18). 

Ideally the hdc-api would return a 404 if there is no licence and that would be propagated through this service